test/controllers/tmf-apis/catalog.js

```diff
@@ -2909,9 +2922,100 @@ describe('Catalog API', function() {
             parentCategoryRequest, { name: categoryName, parentId: parentId, isRoot: false }, 500, errorMsg, done);
     });
 
+    fdescribe('Index in pre validation', function() {
+        var myHelper = function myHelper(done, sName, results, url, pathname, query, expectedUrl, expectedQuery) {
+            var indexes = {};
+            indexes[sName] = q => {
+                if (expectedQuery) {
+                    expect(q).toEqual(expectedQuery);
+                }
+
+
+                return Promise.resolve({
+                    hits: results.map(x => ({document: {originalId: x}}))
+                });
+            };
+
+            var catalogApi = getCatalogApi({}, {}, {}, {}, indexes);
+            var req = {
+                method: "GET",
+                apiUrl: url,
+                _parsedUrl: {
+                    pathname: pathname
+                },
+                query: query
+            };
+
+            catalogApi.checkPermissions(req, function() {
+                expect(req.apiUrl).toEqual(expectedUrl);
+                done();
+            });
+        };
+
+        it('should change catalogs', function(done) {
```

Comment:
I would expect a more descriptive name for the tests, 'should change request URL to include catalog ids when the related party is provided''